cmd/internal/cli/pull.go

```diff
@@ -223,160 +206,46 @@ func pullRun(cmd *cobra.Command, args []string) {
-				sylog.Fatalf("image file already exists: %q - will not overwrite", name)
-			}
-		}
-
 		handlePullFlags(cmd)
 
-		libraryImage, err := client.GetImage(PullLibraryURI, authToken, args[i])
-		if err != nil {
-			sylog.Fatalf("While getting image info: %v", err)
-		}
-
-		var imageName string
-		if transport == "" {
-			imageName = uri.GetName("library://" + args[i])
-		} else {
-			imageName = uri.GetName(args[i])
-		}
-		imagePath := cache.LibraryImage(libraryImage.Hash, imageName)
-		exists, err := cache.LibraryImageExists(libraryImage.Hash, imageName)
-		if err != nil {
-			sylog.Fatalf("unable to check if %v exists: %v", imagePath, err)
-		}
-		if !exists {
-			sylog.Infof("Downloading library image")
-			if err = client.DownloadImage(imagePath, args[i], PullLibraryURI, true, authToken); err != nil {
-				sylog.Fatalf("unable to Download Image: %v", err)
-			}
-
-			if cacheFileHash, err := client.ImageHash(imagePath); err != nil {
-				sylog.Fatalf("Error getting ImageHash: %v", err)
-			} else if cacheFileHash != libraryImage.Hash {
-				sylog.Fatalf("Cached File Hash(%s) and Expected Hash(%s) does not match", cacheFileHash, libraryImage.Hash)
-			}
-		}
-
-		// Perms are 777 *prior* to umask
-		dstFile, err := os.OpenFile(name, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0777)
-		if err != nil {
-			sylog.Fatalf("%v\n", err)
-		}
-		defer dstFile.Close()
-
-		srcFile, err := os.OpenFile(imagePath, os.O_RDONLY, 0444)
-		if err != nil {
-			sylog.Fatalf("%v\n", err)
-		}
-		defer srcFile.Close()
-
-		// Copy SIF from cache
-		_, err = io.Copy(dstFile, srcFile)
-		if err != nil {
-			sylog.Fatalf("%v\n", err)
-		}
-
-		// check if we pulled from the library, if so; is it signed?
-		if !unauthenticatedPull {
-			imageSigned, err := signing.IsSigned(name, KeyServerURL, 0, false, authToken, true)
-			if err != nil {
-				// err will be: "unable to verify container: %v", err
-				sylog.Warningf("%v", err)
-				// if theres a warning, exit 1
-				exitStat = 1
-			}
-			// if container is not signed, print a warning
-			if !imageSigned {
-				fmt.Fprintf(os.Stderr, "This image is not signed, and thus its contents cannot be verified.\n")
-				resp, err := sypgp.AskQuestion("Do you with to proceed? [N/y] ")
-				if err != nil {
-					sylog.Fatalf("unable to parse input: %v", err)
-				}
-				if resp == "" || resp != "y" && resp != "Y" {
-					fmt.Fprintf(os.Stderr, "Aborting.\n")
-					err := os.Remove(name)
-					if err != nil {
-						sylog.Fatalf("Unabel to delete the container: %v", err)
-					}
-					// exit status 10 after replying no
-					exitStat = 10
-				}
-			}
-		} else {
-			sylog.Warningf("Skipping container verification")
+		err := singularity.LibraryPull(name, ref, transport, args[i], PullLibraryURI, KeyServerURL, authToken, force, unauthenticatedPull)
+		if err == singularity.ErrLibraryUnsigned {
+			exitStat = 1
+		} else if err == singularity.ErrLibraryPullAbort {
+			exitStat = 10
+		} else if err != nil {
+			sylog.Fatalf("%v", err)
```

Comment:
Does this need a message? Like:

```suggestion
			sylog.Fatalf("Unable to pull contianer: %v", err)
```